packages/state-transition/src/cache/epochCache.ts

```diff
@@ -529,6 +550,20 @@ export class EpochCache {
     // ```
     this.epoch = computeEpochAtSlot(state.slot);
     this.syncPeriod = computeSyncPeriodAtEpoch(this.epoch);
+
+    // To populate finalized cache and prune unfinalized cache with validators that just entered the activation queue
```

Comment:
ok so strategy here is till the validators are activated, they don't get into finalized cache? which is reasonable sort of even if they could have been added much before at the finalization of a state